client/lib/user/user.js

```diff
@@ -254,26 +254,46 @@ User.prototype.sendVerificationEmail = function( fn ) {
 };
 
 User.prototype.set = function( attributes ) {
-	var changed = false,
-		computedAttributes = userUtils.getComputedAttributes( attributes );
+	let changed = false;
 
-	attributes = Object.assign( {}, attributes, computedAttributes );
-
-	for ( var prop in attributes ) {
-		if ( attributes.hasOwnProperty( prop ) && ! isEqual( attributes[ prop ], this.data[ prop ] ) ) {
-			this.data[ prop ] = attributes[ prop ];
+	for ( const [ attrName, attrValue ] of entries( attributes ) ) {
```

Comment:
Note that ie does not support `for ... of` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

Reply:
The `for ... of` is a part of ES6, so Babel will transpile it to ES5.

Reply:
Can you verify? We don't have full polyfills for other things like missing prototype methods, eg Array.indexOf

Reply:
Yes, it's a language construct, not a method of some object, and it's already used at several places in Calypso.

I couldn't use `Object.entries`, because it's a fairly new thing and we don't polyfill it. That's why I use `lodash.entries`. But `for ... of` is safe.